quadriga/src/main/java/edu/asu/diging/quadriga/core/service/impl/MappedTripleGroupServiceImpl.java

```diff
@@ -0,0 +1,137 @@
+package edu.asu.diging.quadriga.core.service.impl;
+
+import org.bson.types.ObjectId;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.quadriga.core.data.MappedTripleGroupRepository;
+import edu.asu.diging.quadriga.core.exceptions.CollectionNotFoundException;
+import edu.asu.diging.quadriga.core.exceptions.InvalidObjectIdException;
+import edu.asu.diging.quadriga.core.exceptions.MappedTripleGroupNotFoundException;
+import edu.asu.diging.quadriga.core.model.Collection;
+import edu.asu.diging.quadriga.core.model.MappedTripleGroup;
+import edu.asu.diging.quadriga.core.model.MappedTripleType;
+import edu.asu.diging.quadriga.core.service.CollectionManager;
+import edu.asu.diging.quadriga.core.service.MappedTripleGroupService;
+
+@Service
+public class MappedTripleGroupServiceImpl implements MappedTripleGroupService {
+
+    @Autowired
+    private MappedTripleGroupRepository mappedTripleGroupRepository;
+
+    @Autowired
+    private CollectionManager collectionManager;
+
+    /**
+     * This method checks whether a collection with given collectionId exists and
+     * returns the collection if it exists If one doesn't exist, then the
+     * collectionId shouldn't be linked to the mappedTripleGroup
+     * 
+     * @param collectionId is the id of the collection to be checked
+     * @return the Collection entry found in the database
+     * @throws InvalidObjectIdException    if collectionId couldn't be conveted to
+     *                                     ObjectId
+     * @throws CollectionNotFoundException if collection with given collectionId
+     *                                     does't exist
+     */
+    private Collection checkAndGetCollection(String collectionId)
```

Comment:
why is this not just called `get` or `find` or `getCollection`?

Reply:
wait, and why is this method here? Why not use the collection manager?

Reply:
Hi Julia,
I added this method here because it is supposed to throw an exception if a collection is not found because it means that a network is being submitted to a collection that doesn't exist
CollectionManager already has a findCollection method that simply finds and returns the collection and doesn't throw an exception
So I can't rename it to findCollection, probably I'll rename it to getCollection and add it to the CollectionManager, but I'm afraid that getCollection and findCollection, can be confusing

Reply:
I'm getting very  confused with t he different PRs.  Here is what I said on the other one:
why don't you use the collection manager and then check for null instead of creating its own method? This seems unnecessarily complicated.